File: app/src/main/java/code/name/monkey/retromusic/ui/adapter/playlist/AddToPlaylist.java
package code.name.monkey.retromusic.ui.adapter.playlist;

import android.app.Activity;
import android.app.Dialog;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

import code.name.monkey.retromusic.model.Playlist;
import code.name.monkey.retromusic.model.Song;
import code.name.monkey.retromusic.ui.adapter.base.MediaEntryViewHolder;
import code.name.monkey.retromusic.ui.adapter.playlist.AddToPlaylist.ViewHolder;
import code.name.monkey.retromusic.util.PlaylistsUtil;

/**
 * @author Hemanth S (h4h13).
 */
public class AddToPlaylist extends RecyclerView.Adapter<ViewHolder> {

    private Activity activity;
    private ArrayList<Playlist> playlists;
    private int itemLayoutRes;
    private ArrayList<Song> songs;
    private Dialog dialog;

    public AddToPlaylist(Activity activity,
                         ArrayList<Playlist> playlists, int itemLayoutRes,
                         ArrayList<Song> songs, Dialog dialog) {
        this.activity = activity;
        this.playlists = playlists;
        this.itemLayoutRes = itemLayoutRes;
        this.songs = songs;
        this.dialog = dialog;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(activity).inflate(itemLayoutRes, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Playlist playlist = playlists.get(position);
        if (holder.title != null) {
            holder.title.setText(playlist.name);
        }
    }

    @Override
    public int getItemCount() {
        return playlists.size();
    }

    public class ViewHolder extends MediaEntryViewHolder {

        public ViewHolder(View itemView) {
            super(itemView);
        }

        @Override
        public void onClick(View v) {
            super.onClick(v);
            PlaylistsUtil.addToPlaylist(activity, songs, playlists.get(getAdapterPosition()).id, true);
            dialog.dismiss();
        }
    }
}

File: app/src/main/java/code/name/monkey/retromusic/adapter/playlist/AddToPlaylist.kt
package code.name.monkey.retromusic.adapter.playlist

import android.app.Activity
import android.app.Dialog
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import java.util.ArrayList
import androidx.recyclerview.widget.RecyclerView
import code.name.monkey.retromusic.model.Playlist
import code.name.monkey.retromusic.model.Song
import code.name.monkey.retromusic.adapter.base.MediaEntryViewHolder
import code.name.monkey.retromusic.adapter.playlist.AddToPlaylist.ViewHolder
import code.name.monkey.retromusic.util.PlaylistsUtil

class AddToPlaylist(private val activity: Activity,
                    private val playlists: ArrayList<Playlist>, private val itemLayoutRes: Int,
                    private val songs: ArrayList<Song>, private val dialog: Dialog) : RecyclerView.Adapter<ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(activity).inflate(itemLayoutRes, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val playlist = playlists[position]
        if (holder.title != null) {
            holder.title!!.text = playlist.name
        }
    }

    override fun getItemCount(): Int {
        return playlists.size
    }

    inner class ViewHolder(itemView: View) : MediaEntryViewHolder(itemView) {

        override fun onClick(v: View?) {
            super.onClick(v)
            PlaylistsUtil.addToPlaylist(activity, songs, playlists[adapterPosition].id, true)
            dialog.dismiss()
        }
    }
}
